### PR TITLE
[4.0] Backports

### PR DESCRIPTION
**Pull request by nabobalis**
https://github.com/sunpy/sunpy/pull/6410
https://github.com/sunpy/sunpy/pull/6460
https://github.com/sunpy/sunpy/pull/6461

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -600,8 +600,8 @@ Added/Improved Documentation
 Backwards Incompatible Changes
 ------------------------------
 
-- ``sunpy.instr`` has been depreacted and will be removed in sunpy 3.1 in favour of `sunkit_instruments`.
-  The code that is under ``sunpy.instr`` is imported via `sunkit_instruments` to ensure backwards comparability. (`#4526 <https://github.com/sunpy/sunpy/pull/4526>`__)
+- ``sunpy.instr`` has been depreacted and will be removed in sunpy 3.1 in favour of ``sunkit_instruments``.
```
Comment (dstansby) [inline CHANGELOG.rst:603]: Is the better fix here make sure that `sunkit_instruments` is in intersphinx in `conf.py`?

File: tox.ini [hunk 2]
```diff
@@ -76,7 +76,8 @@ extras =
 commands =
     pip freeze --all --no-input
     oldestdeps: python -c "import astropy.time; astropy.time.update_leap_seconds()"
-    !online-!hypothesis-!figure: {env:PYTEST_COMMAND} {posargs}
+    !online-!hypothesis-!figure-!oldestdeps: {env:PYTEST_COMMAND} {posargs}
+    oldestdeps: {env:PYTEST_COMMAND} {posargs} -n 0 --dist no
```
Comment (dstansby) [inline tox.ini:80]: Do you know why xdist isn't working on oldestdeps for 4.0?

Comment (nabobalis) [reply tox.ini:80]: I don't know and this is easier.

File: CHANGELOG.rst (hunk 1 above)
Comment (nabobalis) [reply CHANGELOG.rst:603]: I thought it was, but I didn't check

Comment (dstansby) [reply CHANGELOG.rst:603]: 👍 I think updating `conf.py` is the better fix to do here.

File: tox.ini (hunk 2 above)
Comment (dstansby) [reply tox.ini:80]: 👍 - for future reference can you leave a comment explaining why we're not using `xdist` on oldestdeps, ideally with the error message that happens when it's enabled.

Comment (nabobalis) [reply tox.ini:80]: Will add.

File: CHANGELOG.rst (hunk 1 above)
Comment (nabobalis) [reply CHANGELOG.rst:603]: Its already there. So I don't know why it failed

Comment (dstansby) [reply CHANGELOG.rst:603]: Hmm, it's failing on the normal doc bulids recently too. Maybe a change in `sunkit_instruments`?

Comment (dstansby) [reply CHANGELOG.rst:603]: I think https://github.com/sunpy/sunkit-instruments/commit/0a76e4d75d5c28069ad5e2507c8a1bbb2198f27b#diff-1ab0170e21712558223705b73269528252b7f678af3241add91078655cbfc6f9L10 might have broke this - I'd leave the broken links as broken links in this PR, then we can backport a fix from `main`.

Comment (nabobalis) [reply CHANGELOG.rst:603]: Oh right, I can add that back

**Review by dstansby — APPROVED** [review]

**Comment by dstansby** [comment]
Thanks for doing all these!

**Pull request merged**